### PR TITLE
Add a `type = "cloglog"` for log minus log survival plots

### PR DESCRIPTION
**Pull request by avishaitsur**
**What changes are proposed in this pull request?**
Add a `type = "cloglog"` for log minus log survival plots to `tidysurvfit()` and `ggsurvfit()`. These plots are widely used to assess the assumption of proportional hazards before running a Cox regression model and are implemented in the `survival` and `survminer` packages. 

**If there is an GitHub issue associated with this pull request, please provide link.**
closes https://github.com/pharmaverse/ggsurvfit/issues/194

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark as complete)

- [x] Ensure all package dependencies are installed by running `renv::install()`
- [x] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [x] If a new function was added, function included in `_pkgdown.yml`
- [x] If a bug was fixed, a unit test was added for the bug check
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Overall code coverage remains >99.5%. Review coverage with `withr::with_envvar(list(CI = TRUE), code = devtools::test_coverage())`. Begin in a fresh R session without any packages loaded. 
- [x] R CMD Check runs without errors, warnings, and notes
- [x] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [x] Update `NEWS.md` with the changes from this pull request under the heading "`# ggsurvfit (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [x] Increment the version number using `usethis::use_version(which = "dev")` 
- [x] Run `usethis::use_spell_check()` again
- [x] Approve Pull Request
- [x] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

**Comment by ddsjoberg** [comment]
Thank you for the PR!! I will likely to get to this next week.

**Comment by ddsjoberg** [comment]
it looks like there are some test failures. can you take a look at the details?

**Comment by ddsjoberg** [comment]
FYI, I have just heard from the maintainer of the survival package that they are planning a release that has breaking changes in ggsurvfit. This means that I will prepping a new release soon. If you want to address the failures in the checks in the next week, we may be able to add this to the next release. Let me know if you have questions. Thanks, Daniel

**Comment by avishaitsur** [comment]
Hi Daniel,

I'm a physician, and my coding/GitHub/package development skills are pretty basic. Despite my best efforts, I am struggling with the remaining test failures. I have managed to address some, but others are beyond my ability to resolve efficiently.

I'm reaching out to see if someone within the community could assist in refining these fixes. I believe the updates would be beneficial, especially for users conducting medical research. Any support or direction on how to move forward with these specific issues would be greatly appreciated.

Thank you again for your patience and for considering these modifications.

File: R/ggsurvfit.R [hunk 1]
```diff
@@ -73,13 +73,17 @@ ggsurvfit <- function(x, type = "survival",
   # prep data to be passed to ggplot() -----------------------------------------
   df <-
     tidy_survfit(x = x, type = type) %>%
+    dplyr::filter(type != "cloglog" | .data$time > 0) %>%
```
Comment (ddsjoberg) [inline R/ggsurvfit.R:76]: @avishaitsur can you explain what the purpose of this addition is? Thank you!

Comment (avishaitsur) [reply R/ggsurvfit.R:76]: The transformation in log minus log survival is `cloglog = function(y) log(-log(y))`.
In time 0, by definition, the survival `y` is 1, and therefore, the result is undefined (log of 1 is 0, and log of -0 is undefined).
For this reason, the transformation and the visual do not apply to time 0.

Comment (ddsjoberg) [reply R/ggsurvfit.R:76]: Of course, thank you!

Comment (ddsjoberg) [reply R/ggsurvfit.R:76]: I wonder the best way to handle this as `survfit()` allows negative times. 🤔

Comment (avishaitsur) [reply R/ggsurvfit.R]: The transformation and the visual are implemented in the `survival` and `survminer` packages. Maybe we could explore their solutions?

Comment (ddsjoberg) [reply R/ggsurvfit.R]: worst case, we get a NaN or Inf value and it won't appear on the plot.

I am going to go ahead and merge this now (assuming the current tests complete without error). I am also going to make a release

**Comment by ddsjoberg** [comment]
FYI I removed the call to `ggplot2::scale_x_log10` because we were mixing log base *e* with log base 10. I added an example to the gallery using the correct base log.

**Pull request merged**